### PR TITLE
feat(code-snippets): linked or unlinked templated components

### PR DESCRIPTION
**Pull request by christine-koul**
These changes add code snippets for the following scenarios:

1. PAEngineMultipleUnitLinkedTemplatedComponent
2. PAEngineMultipleUnitUnlinkedTemplatedComponent
3. PAEngineSingleUnitLinkedTemplatedComponent
4. PAEngineSingleUnitUnlinkedTemplatedComponent

jira: https://bpm.factset.com/browse/AAPI-2417

### CODE REVIEW

File: examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineMultipleUnitLinkedTemplatedComponentExample.cs [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+            var linkedPATemplateParametersRoot = new LinkedPATemplateParametersRoot(linkedPATemplateParameters);
+            return linkedPATemplateParametersRoot;
+        }
+
+        private static TemplatedPAComponentParametersRoot GetTemplatedPAComponentParametersRoot(
+            List<PAIdentifier> paAccounts, List<PAIdentifier> paBenchmarks,
+            PADateParameters paDates, List<PACalculationColumn> columns, List<PACalculationGroup> groups, string parentTemplateId)
+        {
+            var templatedPAComponentData =
+                new PAComponentData(paAccounts, paBenchmarks, groups, columns, paDates, CurrencyISOCode, ComponentDetail);
```
Comment (saigiridhar21) [inline examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineMultipleUnitLinkedTemplatedComponentExample.cs:243]: Let's not use a constructor. Can you please use set methods? With new properties, the order of arguments in the constructor can change. Please make this change for both python and dotnet PRs.

Comment (christine-koul) [reply examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineMultipleUnitLinkedTemplatedComponentExample.cs]: fixed https://github.com/factset/analyticsapi-engines-dotnet-sdk/pull/68/commits/f76c9f2dc71e65ea5e25734ca8d408f49ffe4e75

**Review by saigiridhar21 — APPROVED** [review]

File: examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineMultipleUnitLinkedTemplatedComponentExample.cs [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+                    status = getStatusResponse.Data;
+                }
+                Console.WriteLine("Calculation Completed");
+
+                
+                foreach (var calculation in status.Data.Units)
+                {
+                    if (calculation.Value.Status == CalculationUnitStatus.StatusEnum.Success)
+                    {
+                        var resultResponse = calculationApi.GetCalculationUnitResultByIdWithHttpInfo(id: calculationId, unitId: calculation.Key);
```
Comment (vamshimupparaju) [inline examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineMultipleUnitLinkedTemplatedComponentExample.cs:99]: Print success status also same as failure status (example : Calculation Id with unit id - status)

File: examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineMultipleUnitUnlinkedTemplatedComponentExample.cs [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+                    status = getStatusResponse.Data;
+                }
+                Console.WriteLine("Calculation Completed");
+
+                
+                foreach (var calculation in status.Data.Units)
+                {
+                    if (calculation.Value.Status == CalculationUnitStatus.StatusEnum.Success)
+                    {
+                        var resultResponse = calculationApi.GetCalculationUnitResultByIdWithHttpInfo(id: calculationId, unitId: calculation.Key);
```
Comment (vamshimupparaju) [inline examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineMultipleUnitUnlinkedTemplatedComponentExample.cs:97]: Display success status with calc id and unit id to the console.

File: examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineMultipleUnitLinkedTemplatedComponentExample.cs [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+
+            _engineApiConfiguration = new Configuration
+            {
+                BasePath = BasePath,
+                Username = UserName,
+                Password = Password
+            };
+            
+            // Uncomment below lines for adding the proxy configuration
+            //System.Net.WebProxy webProxy = new System.Net.WebProxy("http://myProxyUrl:80/");
```
Comment (vamshimupparaju) [inline examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineMultipleUnitLinkedTemplatedComponentExample.cs:140]: Set Proxy URL as environment variable and read proxy URL also from environment variable.

File: examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineMultipleUnitLinkedTemplatedComponentExample.cs (hunk 2 above)
Comment (christine-koul) [reply examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineMultipleUnitLinkedTemplatedComponentExample.cs:100]: done https://github.com/factset/analyticsapi-engines-dotnet-sdk/pull/68/commits/abcdd14a92a497c5aa6684f882b5a18fe5ca6004

File: examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineMultipleUnitLinkedTemplatedComponentExample.cs (hunk 4 above)
Comment (christine-koul) [reply examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineMultipleUnitLinkedTemplatedComponentExample.cs]: done https://github.com/factset/analyticsapi-engines-dotnet-sdk/pull/68/commits/abcdd14a92a497c5aa6684f882b5a18fe5ca6004

**Review by vamshimupparaju — APPROVED** [review]
Looks good to me

**Pull request merged**